externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -169,6 +168,8 @@ object SparkSQLEngine extends Logging {
 
     if (Utils.isOnK8s) {
       kyuubiConf.setIfMissing(FRONTEND_CONNECTION_URL_USE_HOSTNAME, false)
+    } else {
+      _sparkConf.setIfMissing("spark.ui.port", "0")
```

Comment:
this may cause regression if the user deploys Kyuubi on K8s and run Spark in Client mode, but I guess this is rare and acceptable? WDYT @iodone @yaooqinn

Reply:
let's narrow the condition to "!k8s_cluster_mode"